consensus/serde_utils/src/u256_hex_be.rs

```diff
@@ -0,0 +1,115 @@
+use ethereum_types::U256;
+
+use crate::u64_hex_be::QuantityVisitor;
+use serde::de::Error;
+use serde::{Deserializer, Serializer};
+
+const BYTES_LEN: usize = 32;
+
+pub fn serialize<S>(num: &U256, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let mut bytes = [0; BYTES_LEN];
+    num.to_big_endian(&mut bytes);
+    let raw = hex::encode(bytes);
+    let trimmed = raw.trim_start_matches('0');
+
+    let hex = if trimmed.is_empty() { "0" } else { trimmed };
+
+    serializer.serialize_str(&format!("0x{}", &hex))
+}
+
+pub fn deserialize<'de, D>(deserializer: D) -> Result<U256, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let decoded = deserializer.deserialize_str(QuantityVisitor)?;
+
+    // TODO: this is not strict about byte length like other methods.
+    if decoded.len() > BYTES_LEN {
+        return Err(D::Error::custom(format!(
+            "expected max {} bytes for array, got {}",
+            BYTES_LEN,
+            decoded.len()
+        )));
+    }
+
+    let mut array = [0; BYTES_LEN];
+    array[BYTES_LEN - decoded.len()..].copy_from_slice(&decoded);
```

Comment:
Instead of checking the length on line 30 and then doing the subtraction here, you could convince the compiler that this will never panic with something like:

```rust
let start = BYTES_LEN.checked_sub(decoded.len())
  .ok_or_else(|| {
      return Err(D::Error::custom(format!(
            "expected max {} bytes for array, got {}",
            BYTES_LEN,
            decoded.len()
        )));
  })?;
```

Reply:
I took your advice and refactored it to use the existing U256 serializer and deserializer.